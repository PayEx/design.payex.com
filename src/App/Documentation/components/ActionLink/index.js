import React from "react";

import { ComponentPreview, DocContainer } from "#";

const Overview = () => (
    <>
        <h2 id="overview">Overview</h2>
        <p>Action links.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <a className="action-link" href="#">{"\n"}
                My link{"\n"}
            </a>{"\n"}
        </ComponentPreview>
    </>
);

const ActionLinkBadge = () => (
    <>
        <h2 id="action-link-badge">Action Link with Badge</h2>
        <p>Action links.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <a className="action-link" href="#">{"\n"}
                <span className="badge badge-brand">40</span> {"\n"}
                My link{"\n"}
            </a>{"\n"}
        </ComponentPreview>
    </>
);

const ActionLink = () => (
    <DocContainer docToc>
        <p className="lead">
            Action link
        </p>
        <Overview />
        <ActionLinkBadge />
    </DocContainer>
);

export default ActionLink;

/* For testing */
export { Overview, ActionLinkBadge };
