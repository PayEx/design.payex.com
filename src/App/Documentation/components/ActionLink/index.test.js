import React from "react";
import { shallow } from "enzyme";

import ActionLink, { Overview, ActionLinkBadge } from "./index";

describe("Components: ActionLink", () => {
    it("is defined", () => {
        expect(ActionLink).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<ActionLink />);

        expect(wrapper).toMatchSnapshot();
    });

    describe("Overview", () => {
        it("is defined", () => {
            expect(Overview).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<Overview />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("ActionLinkBadge", () => {
        it("is defined", () => {
            expect(ActionLinkBadge).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<ActionLinkBadge />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
