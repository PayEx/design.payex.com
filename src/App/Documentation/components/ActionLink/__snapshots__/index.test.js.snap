// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: ActionLink ActionLinkBadge renders 1`] = `
<Fragment>
  <h2
    id="action-link-badge"
  >
    Action Link with Badge
  </h2>
  <p>
    Action links.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <a
      className="action-link"
      href="#"
    >
      

      <span
        className="badge badge-brand"
      >
        40
      </span>
       
      

      My link
      

    </a>
    

  </ComponentPreview>
</Fragment>
`;

exports[`Components: ActionLink Overview renders 1`] = `
<Fragment>
  <h2
    id="overview"
  >
    Overview
  </h2>
  <p>
    Action links.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <a
      className="action-link"
      href="#"
    >
      

      My link
      

    </a>
    

  </ComponentPreview>
</Fragment>
`;

exports[`Components: ActionLink renders 1`] = `
<DocContainer
  docToc={true}
>
  <p
    className="lead"
  >
    Action link
  </p>
  <Overview />
  <ActionLinkBadge />
</DocContainer>
`;
